### ISSUE TITLE
Can't get Tiled to run on Ubuntu 22.04.01LTS

### ISSUE THREAD

**Issue by electronmage** [mapeditor/tiled#3503]
After installing Tiled, I can get the following by running 

     snap info tiled

showing that it is installed.

```
name:      tiled
summary:   Your free, easy to use and flexible tile map editor.
publisher: Thorbjørn Lindeijer (bjorn)
store-url: https://snapcraft.io/tiled
contact:   https://discourse.mapeditor.org
license:   GPL-2.0
description: |
  Tiled is a free software level editor. It supports editing tile maps in
  various projections (orthogonal, isometric, hexagonal) and also supports
  building levels with freely positioned, rotated or scaled images or
  annotating them with objects of various shapes.
commands:
  - tiled.terraingenerator
  - tiled
  - tiled.tmxrasterizer
  - tiled.tmxviewer
snap-id:      rqLaGwUxwsLj0suadyXMIuue1Vv5SuaJ
tracking:     latest/stable
refresh-date: 2 days ago, at 15:09 EDT
channels:
  latest/stable:    1.9.2 2022-10-17 (3993) 113MB -
  latest/candidate: 1.9.2 2022-09-16 (3993) 113MB -
  latest/beta:      1.9.2 2022-10-17 (3993) 113MB -
  latest/edge:      1.9.2 2022-10-17 (3993) 113MB -
installed:          1.9.2            (3993) 113MB -
```

But when running:

    tiled

from the terminal or clicking on the Tiled icon or running tiled --version, I get the following:

> Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
> qt.qpa.plugin: Could not find the Qt platform plugin "wayland-egl" in ""
> This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
> 
> Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.
> 
> Aborted (core dumped)

Any ideas?

**Comment by bjorn** [comment]
@electronmage Hmm, so it appears to be trying to find a platform plugin called "wayland-egl", but this is currently not added to the snap. I could look into adding it, but in the meantime, maybe try setting `QT_QPA_PLATFORM=xcb` before running Tiled:
```
export QT_QPA_PLATFORM=xcb
tiled
```
This should cause Tiled to run using X11, which should anyway have less issues than native Wayland.

**Comment by electronmage** [comment]
Thank you for replying.

It just started working.  I didn't do anything additional, but three days
after I installed... It worked!.

I love it.

**Comment by bjorn** [comment]
> It just started working. I didn't do anything additional, but three days after I installed... It worked!.

I've tested running the snap inside of Weston and it didn't work for me with the missing plugins. In the above change I've made sure the Wayland plugins are shipped in the snap release, which fixed the problem for me.

In Tiled 1.10.1, I've additionally updated the snap from core20 to core22, which should make it work a bit better on Wayland due to the upgrade to Qt 5.15. You might still have a better experience running Tiled through an X11 compatibility layer though.
